metagraph/src/graph/representation/succinct/dbg_succinct.cpp

```diff
@@ -270,27 +270,17 @@ void DBGSuccinct::map_to_nodes_sequentially(std::string_view sequence,
     );
 }
 
-void DBGSuccinct
-::call_nodes_with_suffix_matching_longest_prefix(
-            std::string_view str,
-            std::function<void(node_index, uint64_t /* match length */)> callback,
-            size_t min_match_length,
-            size_t max_num_allowed_matches) const {
-    if (!max_num_allowed_matches)
-        return;
-
+std::tuple<boss::BOSS::edge_index, boss::BOSS::edge_index, size_t>
```

Comment:
size + range (begin, end)?

Reply:
I added a comment describing its output. It's more convenient to return the indices of the last outgoing edge from the first and last BOSS node in the range (since `BOSS::tighten_range` can be called on it directly).

Reply:
Though I'm thinking that this would probably be better suited as a BOSS function since it doesn't really use anything specific to DBGSuccinct.

Reply:
Sounds good, seems like you operate on the level of the boss table here.